website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md

```diff
@@ -0,0 +1,410 @@
+
+Store the following script as `create-secret.sh`.
+
+```text
+#!/bin/sh
+
+# Generate a random password
+PASSWORD="$(dd if=/dev/urandom bs=1 count=12 status=none | base64 | tr -dc 'a-zA-Z0-9' | head -c 12)"
+
+kubectl create secret generic ferretuser-secret \
+ --from-literal=username=ferretuser \
+ --from-literal=password="${PASSWORD}" \
+ --type=kubernetes.io/basic-auth \
+ --namespace=cnpg
+```
+
+To make the script file executable, run `chmod +x create-secret.sh` from the file directory in terminal.
+
+Then execute the script to create the `Secret` in the `cnpg` namespace:
+
+```sh
+./create-secret.sh
+```
+
+Confirm the `Secret` was created successfully by running `kubectl get secrets -n cnpg`.
+
+### Deploy Postgres cluster
+
+The `yaml` file below will create a Postgres cluster named "postgres-cluster" in the "cnpg" namespace with three instances, 1Gi storage, and initializes a database named "ferretdb" owned by "ferretuser" using credentials from the "ferretuser-secret".
+
+```yaml
+apiVersion: postgresql.cnpg.io/v1
+kind: Cluster
+metadata:
+  name: postgres-cluster
+  namespace: cnpg
+spec:
+  instances: 3
+  storage:
+    size: 1Gi
+  bootstrap:
+    initdb:
+      database: ferretdb
+      owner: ferretuser
+      secret:
+        name: ferretuser-secret
+```
+
+Apply the `yaml` file:
+
+```sh
+kubectl apply -f pg-cluster.yaml
+```
+
+It may take a while for all the resources to be created.
+Run `kubectl get pods -n cnpg` to be sure all pods are initialized and running.
+
+The output should look like this:
+
+```text
+NAME                                   READY   STATUS    RESTARTS   AGE
+cnpg-cloudnative-pg-6685f6c7b8-5w8pp   1/1     Running   0          11m
+postgres-cluster-1                     1/1     Running   0          116s
+postgres-cluster-2                     1/1     Running   0          97s
+postgres-cluster-3                     1/1     Running   0          79s
+```
+
+## Deploy FerretDB to connect to the Postgres cluster
+
+Now that Postgres cluster has been deployed, let's set up and deploy FerretDB within the same namespace.
+Create another `yaml` file named "ferretdeployment.yaml" for the FerretDB deployment.
+
+FerretDB will connect to the Postgres cluster using the `FERRETDB_POSTGRESQL_URL` environment variable.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ferretdb
+  namespace: cnpg
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ferretdb
+  template:
+    metadata:
+      labels:
+        app: ferretdb
+    spec:
+      containers:
+        - name: ferretdb
+          image: ghcr.io/ferretdb/ferretdb:latest
+          ports:
+            - containerPort: 27017
+          env:
+            - name: FERRETDB_POSTGRESQL_URL
+              value: 'postgresql://ferretuser@postgres-cluster-rw.cnpg.svc.cluster.local:5432/ferretdb'
```

Comment:
Why don't we need password in this connection string?

Reply:
Because unauthenticated access is not desired.

Reply:
Ah, got it! we use `PLAIN` authentication in this example, maybe it is worth mentioning it for clarity. 
@chilagrow thanks!